src/main/java/frc/robot/subsystems/DriveSubsystem.java

```diff
@@ -163,6 +165,37 @@ public void periodic() {
 
     SmartDashboard.putNumberArray("AdvantageScope Swerve Desired States", logDataDesired);
     SmartDashboard.putNumberArray("AdvantageScope Swerve States", logData);
+
+    Pose2d flippedPose = AllianceFlipUtil.apply(getPose());
+    m_flippedField.setRobotPose(flippedPose);
+    SmartDashboard.putNumber("flip X", flippedPose.getX());
+    SmartDashboard.putNumber("flip Y", flippedPose.getY());
+    SmartDashboard.putNumber("flip Rot", flippedPose.getRotation().getDegrees());
+  }
+
+  public void measureLimelight(String name, boolean useLimelight) {
+
+    if (!useLimelight) {
+      return;
+    }
+
+    boolean LLreal = LimelightHelpers.getLatency_Pipeline(name) != 0.0;
+    if (VisionConstants.kUseVision && Robot.isReal() && LLreal) {
+      // Update LimeLight with current robot orientation
+      LimelightHelpers.SetRobotOrientation(name, m_poseEstimator.getEstimatedPosition().getRotation().getDegrees(), 0.0, 0.0, 0.0, 0.0, 0.0);
+
+      // Get the pose estimate
+      LimelightHelpers.PoseEstimate limelightMeasurement = LimelightHelpers.getBotPoseEstimate_wpiBlue_MegaTag2(name);
+
+      // Add it to your pose estimator if it is a valid measurement
+      if (limelightMeasurement != null && limelightMeasurement.tagCount != 0 && m_gyro.getRate() < 720) {
+        m_poseEstimator.addVisionMeasurement(
```

Comment:
Does adding the second one as another vision measurement help? This will ultimately just mean the latest one will end up winning, which helps because we have a higher change of one being accurate, but would it be better to average the two values?